apollo-federation/src/merge.rs

```diff
@@ -135,16 +135,11 @@ impl Merger {
             .map(|(_, subgraph)| subgraph)
             .collect_vec();
         subgraphs.sort_by(|s1, s2| s1.name.cmp(&s2.name));
-        let mut subgraphs_and_enum_values: Vec<(&ValidFederationSubgraph, Name)> = Vec::new();
+        let mut subgraphs_and_enum_values = Vec::new();
         for subgraph in &subgraphs {
-            // TODO: Implement JS codebase's name transform (which always generates a valid GraphQL
-            // name and avoids collisions).
-            if let Ok(subgraph_name) = Name::new(&subgraph.name.to_uppercase()) {
-                subgraphs_and_enum_values.push((subgraph, subgraph_name));
-            } else {
-                self.errors.push(String::from(
-                    "Subgraph name couldn't be transformed into valid GraphQL name",
-                ));
+            match EnumValue::new(&subgraph.name) {
+                Ok(enum_value) => subgraphs_and_enum_values.push((subgraph, enum_value)),
```

Comment:
Does this need to worry about collisions resulting from the normalization? For example, if two names were the same except for characters that were all converted to "_"?

Reply:
This is only being done for our auto-transformed names, so _hopefully_ that's really unlikely. It'd need to be two subgraphs which have nearly identical names and also have a `@connect` on the same path.

Long term, though, for sure we should also port [this](https://github.com/apollographql/federation/blob/c4744da360235d8bb8270ea048f0e0fa5d03be1e/internals-js/src/specs/joinSpec.ts#L212)